### PR TITLE
rename Log and Logf interface to AddEvent and AddEventf.

### PR DESCRIPTION
**Pull request by rghetia**
- also removed log package.

### CODE REVIEW

File: api/trace/span.go [hunk 1]
```diff
@@ -167,12 +166,24 @@ func (sp *span) Tracer() Tracer {
 	return sp.tracer
 }
 
-func (sp *span) Log(ctx context.Context, msg string, args ...core.KeyValue) {
-	log.With(sp).Log(ctx, msg, args...)
+func (sp *span) AddEvent(ctx context.Context, msg string, attributes ...core.KeyValue) {
```
Comment (bogdandrutu) [inline api/trace/span.go]: See how we solve this in Java. We have an interface called Event that can be implemented as "EventF" or as "EventString" so the main Span Interface does not have to provide all these variants.

Comment (rghetia) [reply api/trace/span.go]: done.

**Comment by rghetia** [comment]
> Thank you. I will be working today on a PR to separate the API from the SDK more completely, as discussed in issue 20.

I have started to separate this morning.

**Comment by rghetia** [comment]
@bogdandrutu PTAL.

File: api/event/event.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+package event
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-go/api/core"
+)
+
+type (
+	event struct {
```
Comment (bogdandrutu) [inline api/event/event.go]: Add that var trick that enforces this struct to implement the Event interface?

Comment (rghetia) [reply api/event/event.go]: done

**Pull request merged**